### PR TITLE
refac(messages): improved text, commands and the way the bot handle rates

### PR DESCRIPTION
**Pull request by joaoscheuermann**
- Improved text overall;
- Added a script to load all commands from a folder;
- Updated how the bot handles the rates for USD, so it doesn't get confused any more;
- Improved the way the bot deals with updates on the rates, to be able to send better messages on the channel;
- Added a slash command `dispatch_market_events` to help testing the events on the Market class, it only works on `DEV` with the env var  `ENABLE_DEVELOPMENT_COMMANDS` set to `true`

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,6 +20,7 @@
     "@types/jest": "^29.5.6",
     "@types/node": "^20.8.2",
     "@types/node-cron": "^3.0.10",
+    "@types/sinon": "^10.0.20",
```
Comment (CaioCS97) [inline package.json:23]: I think this is not being used

File: src/index.ts [hunk 2]
```diff
@@ -39,210 +29,199 @@ const oxr = createOpenExchangeClient(OPEN_EXCHANGE_RATES_API_KEY);
...
-dolinhoCommands.set("ping", {
-  data: new SlashCommandBuilder()
-    .setName('ping')
-    .setDescription('Replies with Pong!'),
-  execute: async (interaction: CommandInteraction) => {
-    await interaction.reply('Pong!');
-  },
+const dolinhoB3Market = new Market();
+
+// Initializes the market with the corretc rates
```
Comment (CaioCS97) [inline src/index.ts:37]: typo

File: src/market/index.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      this.oppenningRate,
+      this.closingRate
+    );
+
+    this.emit(MarketEvents.Closing, { rate, variation, percentage });
+  }
+
+  setCurrentRate(rate: number) {
+    // Since we cant retrieve the data from the last hour, we need to initialize the last rate with the current rate,
+    // so we can calculate the variation and percentage on the first update without getting crazy and irrealistc values.
```
Comment (CaioCS97) [inline src/market/index.ts:62]: typo 
```suggestion
    // so we can calculate the variation and percentage on the first update without getting crazy and unrealistic values.
```

File: src/utils/__tests__/calculateVariation.spec.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import { calculateVariation } from "..";
+
+describe("calculateVariation", () => {
+  it("should return [0, 0] when originalValue or currentValue is null", () => {
+    // @ts-ignore
```
Comment (CaioCS97) [inline src/utils/__tests__/calculateVariation.spec.ts:5]: why are these ignores necessary?

**Review by CaioCS97 — CHANGES_REQUESTED** [review]

**Comment by CaioCS97** [comment]
Nice changes!

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (joaoscheuermann) [reply package.json:23]: Done

File: src/discord/commands/usd.ts [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import { DolinhoCommand } from "../../types";
+import { SlashCommandBuilder } from "discord.js";
+
+const command: DolinhoCommand = {
+  dev: true,
+  data: new SlashCommandBuilder()
+    .setName("usd")
+    .setDescription("Replies with Pong!"),
```
[the remark this answers is not in the document]
Comment (joaoscheuermann) [reply src/discord/commands/usd.ts:8]: Done

File: src/index.ts (hunk 2 above)
Comment (joaoscheuermann) [reply src/index.ts:37]: fixed

File: src/utils/__tests__/calculateVariation.spec.ts (hunk 4 above)
Comment (joaoscheuermann) [reply src/utils/__tests__/calculateVariation.spec.ts:5]: Dunno :c chat gpt created it

File: src/market/index.ts (hunk 3 above)
Comment (joaoscheuermann) [reply src/market/index.ts]: Fixed

**Review by CaioCS97 — APPROVED** [review]

**Pull request merged**